[WIP] Summary report

lelemm opened the pull request:
A way to show only a sum of the filter applied to the card.

https://github.com/user-attachments/assets/6c744e9e-10b3-4546-adce-7f446f2464be



Possible new modes:
- Show it as an average per transaction (on the filter)
- Show it as an average per month (on the filter)
- Sum of value of the filter / divisor made by another filter. Example: Expenses in restaurants / total of income

lelemm commented:
Added those:

- Show it as an average per transaction (on the filter)
- Show it as an average per month (on the filter)
- Sum of value of the filter / divisor made by another filter. Example: Expenses in restaurants / total of income

Teprifer commented:
I've had a quick play since I saw it discussed in discord.

I'm not seeing any transaction list pop up.

I think mode should show up above the big figure so all the options are together.

Some more info needed around the mode to be clearer what it's displaying or how it's calculated. 
Sum = Net? Looks like so, suggest changing the word to be aligned with the custom reports.
While it can be done by filters, I think some people might find it useful if the modes had spend/income/net variations? This could auto-set the filter the same way the category selection does in custom reports.

I'm not sure what to make of the percentage calculation (live budget used, the payee is <$5/week), I *think* it's doing the reverse calculation of what I'm thinking? That is it's showing my income as a percentage of this payee, not this payee as a percentage of my income? Or is it a percentage of something else? Also not sure why it'd be red.:

![image](https://github.com/user-attachments/assets/ada886fe-e648-4219-b5e1-2da10b179f69)

lelemm commented:
> I'm not seeing any transaction list pop up.

it was removed to open space for the divisor filter


> I think mode should show up above the big figure so all the options are together.

it was on the footer so the number gets centered on the screen. If a lot of filters is made on the top, they keep pushing the number down and smaller.

> Some more info needed around the mode to be clearer what it's displaying or how it's calculated. Sum = Net? Looks like so, suggest changing the word to be aligned with the custom reports. While it can be done by filters, I think some people might find it useful if the modes had spend/income/net variations? This could auto-set the filter the same way the category selection does in custom reports.

it can't be called NET, because its just a sum of the filters allied on the top of the page. If the users select > 0, it just incomes for example

> 
> I'm not sure what to make of the percentage calculation (live budget used, the payee is <$5/week), I _think_ it's doing the reverse calculation of what I'm thinking? That is it's showing my income as a percentage of this payee, not this payee as a percentage of my income? Or is it a percentage of something else? Also not sure why it'd be red.:
> 
> ![image](https://private-user-images.githubusercontent.com/34882928/383467729-ada886fe-e648-4219-b5e1-2da10b179f69.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4ODk4MTUsIm5iZiI6MTczMDg4OTUxNSwicGF0aCI6Ii8zNDg4MjkyOC8zODM0Njc3MjktYWRhODg2ZmUtZTY0OC00MjE5LWI1ZTEtMmRhMTBiMTc5ZjY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA2VDEwMzgzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEzYjVmMGI3MWU5ODRmNzVmYWYzZDY0ZDA1ZjE2ZGNlYzdkNTNhMGNjNTQ1MDgwZjRhOWE0OTE4Mjc4ZDdkODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nflO6dChZZY3gUrEfDZKX561Zbzby-FGRSH-tBYcfwg)

you applied the payee filter on the bottom. So basically what this mean is:
sum of everything / sum of specfic payee = percentage.
if want income over payee, you have to filter above the payee and bellow the categories of income.

this way it will be sum of every transaction for payee / sum of all income

lelemm commented on packages/loot-core/src/types/models/dashboard.d.ts:
```diff
@@ -88,3 +89,21 @@ export type ExportImportDashboard = {
   widgets: ExportImportDashboardWidget[];
 };
+
+export type SummaryWidget = AbstractWidget<
+  'summary-card',
+  {
+    name?: string;
+    conditions?: RuleConditionEntity[];
+    conditionsOp?: 'and' | 'or';
+    timeFrame?: TimeFrame;
+    content?: string;
+  } | null
+>;
```
can't do this, the content is serialized

MikesGlitch reviewed (commented):
Privacy mode isn't hiding the values

![image](https://github.com/user-attachments/assets/7c6a58b3-c9cb-43b6-be89-4be1e1d3f666)

I wonder if we can add a little more padding to the left and right of the number? It looks a little to squeezed in, can you bring it in line with the title?

![image](https://github.com/user-attachments/assets/93872d88-a30f-44f4-ac45-118c81b51ab7)

I also just notice the title stands out a little more than the other cards: 
![image](https://github.com/user-attachments/assets/71cf8a91-9d11-4749-ab04-7beead8bf32a)

Teprifer commented:
Hrm, I still think it's not a good idea to have the settings split apart the way they are, overall it's not intuitive how to get a specific result, or what the result being presented is. This might be a result of using existing mechanisms instead of something specific to this type of report. It's powerful, but not friendly.

While it's ok, there's also no reason the area holding the number has to be so large in the full view, it could be around the size of the default card side, this would also allow space to return the transaction list (or even side by side lists if doing a percentage to show what's on both sides of the equation).

Trying to think of how it could be improved without suggesting whole sale changes.

Ideally if possible, the mode at the bottom needs to be up at the top underneath where it lists filters, so all are together.

Mode could be renamed "Show as: "

Other than that not sure how to suggest making it easier to use and understand without suggesting removing a lot of functionality which _someone_ might want.

lelemm commented:
> Privacy mode isn't hiding the values
> 
> ![image](https://private-user-images.githubusercontent.com/5285928/385464391-7c6a58b3-c9cb-43b6-be89-4be1e1d3f666.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MjkzMTksIm5iZiI6MTczMTUyOTAxOSwicGF0aCI6Ii81Mjg1OTI4LzM4NTQ2NDM5MS03YzZhNThiMy1jOWNiLTQzYjYtYmU4OS00YmUxZTFkM2Y2NjYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjAxNjU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGRhODI1OTViMDgyZmJkM2Q3ZGY5YzQwYjhkNDJhNmRhZDE0ZjVlMzJkYTAyZGU0M2EyZTAzMGM4Y2M2MTg1ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.HJazqSiwbttLoQsDlZK7OK-BFxdAJs1Epe58kYFRYwk)
> 
added privacy filter

> I wonder if we can add a little more padding to the left and right of the number? It looks a little to squeezed in, can you give it a bit more space from the sides of the card - maybe 20px or so on desktop?
> 
> ![image](https://private-user-images.githubusercontent.com/5285928/385469033-da0106f3-6062-4c71-9102-05a735f73d52.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MjkzMTksIm5iZiI6MTczMTUyOTAxOSwicGF0aCI6Ii81Mjg1OTI4LzM4NTQ2OTAzMy1kYTAxMDZmMy02MDYyLTRjNzEtOTEwMi0wNWE3MzVmNzNkNTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjAxNjU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjM2NTQzNTk5N2ZiYjQ3NjMzMjEwZmE4YjQ2OTg5Y2FkOThkN2YxZGM2Zjg5YWE4OWZmMzA1M2IxYmU0NjFiYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.yQA2VCY31uRWj09Shv3G_amh4tgFhCnfS16rSpCIQ_c)
> 
check now, here looks good

> I also noticed the title stands out a little more than the other cards: ![image](https://private-user-images.githubusercontent.com/5285928/385466173-71cf8a91-9d11-4749-ab04-7beead8bf32a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MjkzMTksIm5iZiI6MTczMTUyOTAxOSwicGF0aCI6Ii81Mjg1OTI4LzM4NTQ2NjE3My03MWNmOGE5MS05ZDExLTQ3NDktYWIwNC03YmVlYWQ4YmYzMmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjAxNjU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTNkYTQyYTFiYTAwYjAwYTc1Y2IxMzdiMGM0YWJiM2I1OGI2OGE3ZWM0Y2RmMDk5OGVjMjdhMzM2NjE0YjA5MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.MMqSNqNNUyx85tJDxWzAZZyJ047zgVP6plABbyyuOnA)

![image](https://github.com/user-attachments/assets/0ad0226e-ae35-4fe8-8721-a961fabfacec)
the problem is the custom report, the summary card is like the other cards

lelemm commented:
> Hrm, I still think it's not a good idea to have the settings split apart the way they are, overall it's not intuitive how to get a specific result, or what the result being presented is. This might be a result of using existing mechanisms instead of something specific to this type of report. It's powerful, but not friendly.
> 
> While it's ok, there's also no reason the area holding the number has to be so large in the full view, it could be around the size of the default card side, this would also allow space to return the transaction list (or even side by side lists if doing a percentage to show what's on both sides of the equation).
> 
> Trying to think of how it could be improved without suggesting whole sale changes.
> 
> Ideally if possible, the mode at the bottom needs to be up at the top underneath where it lists filters, so all are together.
> 
> Mode could be renamed "Show as: "
> 
> Other than that not sure how to suggest making it easier to use and understand without suggesting removing a lot of functionality which _someone_ might want.

changed it how it works

MikesGlitch commented:
What are people's thoughts about the animation when you first view the page? I'm not a fan personally:

![test](https://github.com/user-attachments/assets/76d7b949-1280-46e7-927c-1d36fcfa2948)

I think we turned the animation off for the other charts so they just load in when they're ready.

MikesGlitch commented on upcoming-release-notes/3792.md:
```diff
@@ -0,0 +1,6 @@
+---
+category: Enhancements
```
I'd say this is a feature since it's a new thing

MikesGlitch commented:
I may have found a bug, when I try to divide by 0 it sets the dividend number to 0 instead of its real value

![bug](https://github.com/user-attachments/assets/061af89d-87f5-4663-819c-d979df4b9a7c)

lelemm commented:
> What are people's thoughts about the animation when you first view the page? I'm not a fan personally:
> 
> ![test](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE) [ ![test](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE) ](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE) [ ](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE)
> 
> I think we turned the animation off for the other charts so they were less visually distracting when there were loads of them on the page. They just load in when they're ready.

This effect is the component trying to discover the font size. Not sure why it's bouncy for you. It should increase than stop

lelemm commented:
> I may have found a bug, when I try to divide by 0 it sets the dividend number to 0 instead of its real value (in this case 42.62). I found it a bit confusing.
> 
> ![bug](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ) [ ![bug](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ) ](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ) [ ](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ)

Nice catch. I will check it

lelemm commented on packages/desktop-client/src/components/reports/spreadsheets/summary-spreadsheet.ts:
```diff
@@ -0,0 +1,290 @@
+
+  const divisorValue = divisorData?.data?.[0]?.amount ?? 0;
+
+  const dividend = data.reduce((prev, ac) => prev + (ac?.amount ?? 0), 0);
+  return {
+    total: Math.round(((dividend ?? 0) / (divisorValue ?? 1)) * 10000) / 100,
+    divisor: (divisorValue ?? 0) / 100,
```
The division by zero is ok in this case so it results in infinite

lelemm commented:
> > I may have found a bug, when I try to divide by 0 it sets the dividend number to 0 instead of its real value (in this case 42.62). I found it a bit confusing.
> > ![bug](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ)
> >       
> >         [
> >           
> >         
> >             ![bug](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ)
> >           ](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ)
> >         
> >        [ ![bug](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ) ](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ)
> >       
> >         [
> >           
> >         
> >             ![bug](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ)
> >           ](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ)
> >         
> >        [ ](https://private-user-images.githubusercontent.com/5285928/385944810-061af89d-87f5-4663-819c-d979df4b9a7c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTk0NDgxMC0wNjFhZjg5ZC04N2Y1LTQ2NjMtODE5Yy1kOTc5ZGY0YjlhN2MuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWMxMDg0NGE3YzU1NGVlODRmNjE1N2Q2MGIwNzlkNmJhODc0ZjZiNjM1ZGUwYzY5ZjQyNzNiZTRkMGFjZmUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aKjwHBgMloYVcoycp4HebBiVqVlXdKhSb4h0QzCKylQ)
> 
> Nice catch. I will check it

@MikesGlitch now it does not change the Camry but let the number get into infinity

lelemm commented:
> > What are people's thoughts about the animation when you first view the page? I'm not a fan personally:
> > ![test](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE)
> >       
> >         [
> >           
> >         
> >             ![test](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE)
> >           ](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE)
> >         
> >        [ ![test](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE) ](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE)
> >       
> >         [
> >           
> >         
> >             ![test](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE)
> >           ](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE)
> >         
> >        [ ](https://private-user-images.githubusercontent.com/5285928/385939422-76d7b949-1280-46e7-927c-1d36fcfa2948.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NDAxMzIsIm5iZiI6MTczMTUzOTgzMiwicGF0aCI6Ii81Mjg1OTI4LzM4NTkzOTQyMi03NmQ3Yjk0OS0xMjgwLTQ2ZTctOTI3Yy0xZDM2ZmNmYTI5NDguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMjMxNzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjNjMWVkNjMxNmY0Y2EwZWEzMjhjNTI0NGE1ZGZjODgwOTRiZTY1YjdlNThkYzY1N2U3NWQwNTFjZjljYzgzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TdG0uOYXn5IH1KEG0wNZJFSE7NWFmYKXu613c2sfquE)
> > I think we turned the animation off for the other charts so they were less visually distracting when there were loads of them on the page. They just load in when they're ready.
> 
> This effect is the component trying to discover the font size. Not sure why it's bouncy for you. It should increase than stop

I guess I could save the last font size used, and when the card shows up, use the saved size instead of calculating everytime the dashboard opens

Teprifer commented:
I like this iteration quite a bit!

I don't have the bounciness, just the size zooming up (chrome & firefox).

The only thing I'm not sure about is allowing the 'any' option instead of only allowing 'all'. It could just be what I was playing with is so contrived as to be irrelevant though and there might be a use case?
![image](https://github.com/user-attachments/assets/db371407-8a1f-4f5a-8896-f05330002eaa)

Could you add the user preference number formatting to the percentage fraction values? I see you have it for the result already.
![image](https://github.com/user-attachments/assets/768d755c-be56-490a-9a3b-1517264b8c31)

MikesGlitch commented:
> This effect is the component trying to discover the font size. Not sure why it's bouncy for you. It should increase than stop

I'm using edge. Maybe it's because the number is big?

Another thing I think would be good is to not show the number until we know what it is. Right now it defaults to 0. I have a few reports on my screen, it takes a good second for it to turn to the correct value. 

![bug2](https://github.com/user-attachments/assets/16beed18-338b-4c41-979d-8bb53a225b04)

Can we hide the number until we know the calculation is done? This may reduce flickering or transitions between font size.

lelemm commented:
> I like this iteration quite a bit!
> 
> I don't have the bounciness, just the size zooming up (chrome & firefox).

@MikesGlitch @Teprifer I removed the animation from the card, now its animated only during edit mode, otherwise it snaps the font size

> 
> The only thing I'm not sure about is allowing the 'any' option instead of only allowing 'all'. It could just be what I was playing with is so contrived as to be irrelevant though and there might be a use case?

I would not change remove none of the filter options since it allows any kind of calculation a user would be able to do.
I like the 'any' cuz it allows me to select many categories, one on each filter and its explicit what I'm filtering:

![image](https://github.com/user-attachments/assets/4ab9bf3b-370a-4716-b3e3-6e51def1103e)

Those two filters result on the same value, but I find the first one more clear.


> 
> Could you add the user preference number formatting to the percentage fraction values? I see you have it for the result already. 

Added

lelemm commented:
> > This effect is the component trying to discover the font size. Not sure why it's bouncy for you. It should increase than stop
> 
> I'm using edge. Maybe it's because the number is big?
> 
> Another thing I think would be good is to not show the number until we know what it is. Right now it defaults to 0. I have a few reports on my screen, it takes a good second for it to turn to the correct value.
> 
> ![bug2](https://private-user-images.githubusercontent.com/5285928/386163385-16beed18-338b-4c41-979d-8bb53a225b04.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1ODQyODUsIm5iZiI6MTczMTU4Mzk4NSwicGF0aCI6Ii81Mjg1OTI4LzM4NjE2MzM4NS0xNmJlZWQxOC0zMzhiLTRjNDEtOTc5ZC04YmI1M2EyMjViMDQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTEzMzA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2VjNzVhZTM3MTk0OGE3NzliYTMyY2M1Y2IwM2FlMTRjY2IxYWVjOWMzNGZhNzkwYzEzMWZlOGQyODUyZTU4OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.0iQxDNM7GGkVperMSUxFz3_e1DCxIMVne1GnllQyci4) [ ![bug2](https://private-user-images.githubusercontent.com/5285928/386163385-16beed18-338b-4c41-979d-8bb53a225b04.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1ODQyODUsIm5iZiI6MTczMTU4Mzk4NSwicGF0aCI6Ii81Mjg1OTI4LzM4NjE2MzM4NS0xNmJlZWQxOC0zMzhiLTRjNDEtOTc5ZC04YmI1M2EyMjViMDQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTEzMzA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2VjNzVhZTM3MTk0OGE3NzliYTMyY2M1Y2IwM2FlMTRjY2IxYWVjOWMzNGZhNzkwYzEzMWZlOGQyODUyZTU4OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.0iQxDNM7GGkVperMSUxFz3_e1DCxIMVne1GnllQyci4) ](https://private-user-images.githubusercontent.com/5285928/386163385-16beed18-338b-4c41-979d-8bb53a225b04.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1ODQyODUsIm5iZiI6MTczMTU4Mzk4NSwicGF0aCI6Ii81Mjg1OTI4LzM4NjE2MzM4NS0xNmJlZWQxOC0zMzhiLTRjNDEtOTc5ZC04YmI1M2EyMjViMDQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTEzMzA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2VjNzVhZTM3MTk0OGE3NzliYTMyY2M1Y2IwM2FlMTRjY2IxYWVjOWMzNGZhNzkwYzEzMWZlOGQyODUyZTU4OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.0iQxDNM7GGkVperMSUxFz3_e1DCxIMVne1GnllQyci4) [ ](https://private-user-images.githubusercontent.com/5285928/386163385-16beed18-338b-4c41-979d-8bb53a225b04.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1ODQyODUsIm5iZiI6MTczMTU4Mzk4NSwicGF0aCI6Ii81Mjg1OTI4LzM4NjE2MzM4NS0xNmJlZWQxOC0zMzhiLTRjNDEtOTc5ZC04YmI1M2EyMjViMDQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTEzMzA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2VjNzVhZTM3MTk0OGE3NzliYTMyY2M1Y2IwM2FlMTRjY2IxYWVjOWMzNGZhNzkwYzEzMWZlOGQyODUyZTU4OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.0iQxDNM7GGkVperMSUxFz3_e1DCxIMVne1GnllQyci4)
> 
> Can we hide the number until we know the calculation is done? This may reduce flickering or transitions between font size.

I wasn't able to test this locally, but I added a loading indicator, can you try it out @MikesGlitch ?

MikesGlitch commented:
> I wasn't able to test this locally, but I added a loading indicator, can you try it out @MikesGlitch ?

Looks excellent - much more consistent with the other cards now. 👍

MikesGlitch commented on packages/desktop-client/src/components/reports/reports/Summary.tsx:
```diff
@@ -0,0 +1,565 @@
+            padding: 16,
+          }}
+        >
+          <span style={{ marginRight: 4 }}>Show as</span>
```
Not translated

lelemm replied:
done

MikesGlitch commented on packages/desktop-client/src/components/reports/reports/Summary.tsx:
```diff
@@ -0,0 +1,565 @@
+          <FieldSelect
+            style={{ marginRight: 16 }}
+            fields={[
+              ['sum', 'Sum'],
```
Not translated

lelemm replied:
done

MikesGlitch commented on packages/desktop-client/src/components/reports/reports/Summary.tsx:
```diff
@@ -0,0 +1,565 @@
+                }));
+              }}
+            />{' '}
+            Include summary date range
```
Not translated

lelemm replied:
done

MikesGlitch commented on packages/desktop-client/src/components/reports/reports/Summary.tsx:
```diff
@@ -0,0 +1,565 @@
+            style={{ fontSize: '32px', width: '100%', textAlign: 'center' }}
+          >
+            {type === 'avgPerMonth'
+              ? 'number of months'
```
Not translated

lelemm replied:
done

MikesGlitch commented on packages/desktop-client/src/components/reports/reports/Summary.tsx:
```diff
@@ -0,0 +1,565 @@
+      <View style={{ marginLeft: 16, maxWidth: '220px', marginRight: 16 }}>
+        {(filterObject.conditions?.length ?? 0) === 0 ? (
+          <Text style={{ fontSize: '25px', color: theme.pageTextPositive }}>
+            all transactions
```
Not translated

lelemm replied:
done

youngcw commented:
Maybe instead of the "include summary date range" button you could instead say "all time divisor" or something a bit more clear on what the button is doing.

It might be nice if the divisor could default to income in some way.

This may be for later, but it would be nice if the summary card could get a lot smaller, so you could have a bunch of them in a row

MikesGlitch commented:
Bug: I can't rename the summary card from the edit dashboard screen

![bug](https://github.com/user-attachments/assets/66d54aa3-243c-4f7c-ac40-d194f2761f58)

lelemm commented:
> all time divisor

Done, now the check box must be checked to remove the date range from the divisor

lelemm commented:
> Bug: I can't rename the summary card from the edit dashboard screen
> 
> ![bug](https://private-user-images.githubusercontent.com/5285928/386328692-66d54aa3-243c-4f7c-ac40-d194f2761f58.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MTI5MjcsIm5iZiI6MTczMTYxMjYyNywicGF0aCI6Ii81Mjg1OTI4LzM4NjMyODY5Mi02NmQ1NGFhMy0yNDNjLTRmN2MtYWM0MC1kMTk0ZjI3NjFmNTguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTkzMDI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTAxMjE0ZmZiMWRlMjE4MTY0MjhjMWI0YmNkNzM3NDQ0NDFmZDU1ZDk1MzliNzA4MGQ5Njk2Y2Q4OGU0NTlmMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8BNRoO119Wc1ryMZp_Pa_oOu6lNyH7xF8acRLtISIHM) [ ![bug](https://private-user-images.githubusercontent.com/5285928/386328692-66d54aa3-243c-4f7c-ac40-d194f2761f58.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MTI5MjcsIm5iZiI6MTczMTYxMjYyNywicGF0aCI6Ii81Mjg1OTI4LzM4NjMyODY5Mi02NmQ1NGFhMy0yNDNjLTRmN2MtYWM0MC1kMTk0ZjI3NjFmNTguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTkzMDI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTAxMjE0ZmZiMWRlMjE4MTY0MjhjMWI0YmNkNzM3NDQ0NDFmZDU1ZDk1MzliNzA4MGQ5Njk2Y2Q4OGU0NTlmMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8BNRoO119Wc1ryMZp_Pa_oOu6lNyH7xF8acRLtISIHM) ](https://private-user-images.githubusercontent.com/5285928/386328692-66d54aa3-243c-4f7c-ac40-d194f2761f58.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MTI5MjcsIm5iZiI6MTczMTYxMjYyNywicGF0aCI6Ii81Mjg1OTI4LzM4NjMyODY5Mi02NmQ1NGFhMy0yNDNjLTRmN2MtYWM0MC1kMTk0ZjI3NjFmNTguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTkzMDI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTAxMjE0ZmZiMWRlMjE4MTY0MjhjMWI0YmNkNzM3NDQ0NDFmZDU1ZDk1MzliNzA4MGQ5Njk2Y2Q4OGU0NTlmMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8BNRoO119Wc1ryMZp_Pa_oOu6lNyH7xF8acRLtISIHM) [ ](https://private-user-images.githubusercontent.com/5285928/386328692-66d54aa3-243c-4f7c-ac40-d194f2761f58.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MTI5MjcsIm5iZiI6MTczMTYxMjYyNywicGF0aCI6Ii81Mjg1OTI4LzM4NjMyODY5Mi02NmQ1NGFhMy0yNDNjLTRmN2MtYWM0MC1kMTk0ZjI3NjFmNTguZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTkzMDI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTAxMjE0ZmZiMWRlMjE4MTY0MjhjMWI0YmNkNzM3NDQ0NDFmZDU1ZDk1MzliNzA4MGQ5Njk2Y2Q4OGU0NTlmMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8BNRoO119Wc1ryMZp_Pa_oOu6lNyH7xF8acRLtISIHM)

Fixed. I forgot to pass down to the child component the saved font, it was calculating it every rerender and it was overriding the title name

MikesGlitch commented:
This may be my last nit: can we add some space between the top of the number and the title? It's a little too close when the number is 0.00

![image](https://github.com/user-attachments/assets/9a2e449f-7786-46a2-8e7c-ef91fb8b28c5)

lelemm commented:
> This may be my last nit: The font sizing seems to have trouble sometimes - for me It's a little too close when the number is 0.00
> 
> ![image](https://private-user-images.githubusercontent.com/5285928/386347684-9a2e449f-7786-46a2-8e7c-ef91fb8b28c5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MTQ1ODUsIm5iZiI6MTczMTYxNDI4NSwicGF0aCI6Ii81Mjg1OTI4LzM4NjM0NzY4NC05YTJlNDQ5Zi03Nzg2LTQ2YTItOGU3Yy1lZjkxZmI4YjI4YzUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTk1ODA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWQzMGZkMGZmMGQxZjRiZGFhNzFmYjNjMTE0MTgyYmRkNzczZWE3N2E0Y2IzYzhlOTA4NjI1MzdkYzM5MDc1MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.w5B3FychcRWj08D92_T2xEIQBl6XzXBQzpk63EJQshQ)
> 
> It displayed well for me at first, but when navigating to budget screen, then back to reports it displayed like above.
> 
> I also get this on a small desktop screen - the font doesn't seem to adjust well on small desktops/tablet sizes: ![bug2](https://private-user-images.githubusercontent.com/5285928/386348999-e5fae101-b61a-474d-8733-4bdc32f6aacb.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MTQ1ODUsIm5iZiI6MTczMTYxNDI4NSwicGF0aCI6Ii81Mjg1OTI4LzM4NjM0ODk5OS1lNWZhZTEwMS1iNjFhLTQ3NGQtODczMy00YmRjMzJmNmFhY2IuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTk1ODA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjNjZGQwZTg3ZTI3YTgxNDFmMWNlMzYwN2JkNThlZjg4MzMwNGNiMWNiMzc4M2NkOGI2Mzg5NmI4NDMyNmZiMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8P6mrr3v-wKI56mNmYF5wCf9JpSwqrcGY_MO5kyEFYw) [ ![bug2](https://private-user-images.githubusercontent.com/5285928/386348999-e5fae101-b61a-474d-8733-4bdc32f6aacb.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MTQ1ODUsIm5iZiI6MTczMTYxNDI4NSwicGF0aCI6Ii81Mjg1OTI4LzM4NjM0ODk5OS1lNWZhZTEwMS1iNjFhLTQ3NGQtODczMy00YmRjMzJmNmFhY2IuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTk1ODA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjNjZGQwZTg3ZTI3YTgxNDFmMWNlMzYwN2JkNThlZjg4MzMwNGNiMWNiMzc4M2NkOGI2Mzg5NmI4NDMyNmZiMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8P6mrr3v-wKI56mNmYF5wCf9JpSwqrcGY_MO5kyEFYw) ](https://private-user-images.githubusercontent.com/5285928/386348999-e5fae101-b61a-474d-8733-4bdc32f6aacb.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MTQ1ODUsIm5iZiI6MTczMTYxNDI4NSwicGF0aCI6Ii81Mjg1OTI4LzM4NjM0ODk5OS1lNWZhZTEwMS1iNjFhLTQ3NGQtODczMy00YmRjMzJmNmFhY2IuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTk1ODA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjNjZGQwZTg3ZTI3YTgxNDFmMWNlMzYwN2JkNThlZjg4MzMwNGNiMWNiMzc4M2NkOGI2Mzg5NmI4NDMyNmZiMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8P6mrr3v-wKI56mNmYF5wCf9JpSwqrcGY_MO5kyEFYw) [ ](https://private-user-images.githubusercontent.com/5285928/386348999-e5fae101-b61a-474d-8733-4bdc32f6aacb.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MTQ1ODUsIm5iZiI6MTczMTYxNDI4NSwicGF0aCI6Ii81Mjg1OTI4LzM4NjM0ODk5OS1lNWZhZTEwMS1iNjFhLTQ3NGQtODczMy00YmRjMzJmNmFhY2IuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMTk1ODA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjNjZGQwZTg3ZTI3YTgxNDFmMWNlMzYwN2JkNThlZjg4MzMwNGNiMWNiMzc4M2NkOGI2Mzg5NmI4NDMyNmZiMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8P6mrr3v-wKI56mNmYF5wCf9JpSwqrcGY_MO5kyEFYw)

I will check

lelemm commented:
> This may be my last nit: The font sizing seems to have trouble sometimes - for me It's a little too close when the number is 0.00
> 
> ![image](https://private-user-images.githubusercontent.com/5285928/386347684-9a2e449f-7786-46a2-8e7c-ef91fb8b28c5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2NDEyNTgsIm5iZiI6MTczMTY0MDk1OCwicGF0aCI6Ii81Mjg1OTI4LzM4NjM0NzY4NC05YTJlNDQ5Zi03Nzg2LTQ2YTItOGU3Yy1lZjkxZmI4YjI4YzUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTVUMDMyMjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODljNjAzOGRlZTM0ZWZmZGZkZjc3NzU5ZmU0M2Y4NGRiMDViNjhiNmE0MjRjN2Y5NWI2OTYxMmVjNGI2YzIxYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.tZ3K3bpEG0ErQHCaC-iW__axBIPv9N0TPvuMuPv_3zw)
> 
> It displayed well for me at first, but when navigating to budget screen, then back to reports it displayed like above.
> 
> I also get this on a small desktop screen - the font doesn't seem to adjust well on small desktops/tablet sizes: ![bug2](https://private-user-images.githubusercontent.com/5285928/386348999-e5fae101-b61a-474d-8733-4bdc32f6aacb.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2NDEyNTgsIm5iZiI6MTczMTY0MDk1OCwicGF0aCI6Ii81Mjg1OTI4LzM4NjM0ODk5OS1lNWZhZTEwMS1iNjFhLTQ3NGQtODczMy00YmRjMzJmNmFhY2IuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTVUMDMyMjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWIxOGI4YmMwZmJlMDRhYmZhZmVkYmQ5NzM0YTQ2YWMxNzY0YWUxMDFmOGQ4ODIyMmVhNDdiNTNkNDg3YWMzNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.xMuHP0InMkN73GIwsIJ2L046UlpOVU6S-DctVrsQQ6U) [ ![bug2](https://private-user-images.githubusercontent.com/5285928/386348999-e5fae101-b61a-474d-8733-4bdc32f6aacb.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2NDEyNTgsIm5iZiI6MTczMTY0MDk1OCwicGF0aCI6Ii81Mjg1OTI4LzM4NjM0ODk5OS1lNWZhZTEwMS1iNjFhLTQ3NGQtODczMy00YmRjMzJmNmFhY2IuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTVUMDMyMjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWIxOGI4YmMwZmJlMDRhYmZhZmVkYmQ5NzM0YTQ2YWMxNzY0YWUxMDFmOGQ4ODIyMmVhNDdiNTNkNDg3YWMzNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.xMuHP0InMkN73GIwsIJ2L046UlpOVU6S-DctVrsQQ6U) ](https://private-user-images.githubusercontent.com/5285928/386348999-e5fae101-b61a-474d-8733-4bdc32f6aacb.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2NDEyNTgsIm5iZiI6MTczMTY0MDk1OCwicGF0aCI6Ii81Mjg1OTI4LzM4NjM0ODk5OS1lNWZhZTEwMS1iNjFhLTQ3NGQtODczMy00YmRjMzJmNmFhY2IuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTVUMDMyMjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWIxOGI4YmMwZmJlMDRhYmZhZmVkYmQ5NzM0YTQ2YWMxNzY0YWUxMDFmOGQ4ODIyMmVhNDdiNTNkNDg3YWMzNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.xMuHP0InMkN73GIwsIJ2L046UlpOVU6S-DctVrsQQ6U) [ ](https://private-user-images.githubusercontent.com/5285928/386348999-e5fae101-b61a-474d-8733-4bdc32f6aacb.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2NDEyNTgsIm5iZiI6MTczMTY0MDk1OCwicGF0aCI6Ii81Mjg1OTI4LzM4NjM0ODk5OS1lNWZhZTEwMS1iNjFhLTQ3NGQtODczMy00YmRjMzJmNmFhY2IuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTVUMDMyMjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWIxOGI4YmMwZmJlMDRhYmZhZmVkYmQ5NzM0YTQ2YWMxNzY0YWUxMDFmOGQ4ODIyMmVhNDdiNTNkNDg3YWMzNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.xMuHP0InMkN73GIwsIJ2L046UlpOVU6S-DctVrsQQ6U)

check now

lelemm commented:
> > check now
> 
> That's fixes the issue on smaller screens. I noticed the font is a bit smaller now. It leaves quite a lot of whitespace on the card:
> 
> ![image](https://private-user-images.githubusercontent.com/5285928/386576747-ce20e363-3443-4beb-b976-c26798963913.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5MzUwNDMsIm5iZiI6MTczMTkzNDc0MywicGF0aCI6Ii81Mjg1OTI4LzM4NjU3Njc0Ny1jZTIwZTM2My0zNDQzLTRiZWItYjk3Ni1jMjY3OTg5NjM5MTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMThUMTI1OTAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGQ4NjE5YTM5NmQwZmJjNzhiMDMyMGViNWY2ZjQ1MWQ0YjI1OGU1Yzc3MDkzMDkxMWU5NDAzMjAzZjE1YWRhYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.H047QnYRakJxeZdNJeqjzu2SxNgUqBYa6mYTXihIBxg)
> 
> If this is the price to pay for it working on smaller screens I'm good with it, but ideally the text would fill out the space a little more IMO.

I did some changes. It looks like it fits better. Take a look

lelemm commented:
> It seems that with the newest version, modifying an existing filter creates a copy for the modified one and adds it to the filter stack.

Not sure when I introduced this error, but fixed

lelemm commented:
yeah, I removed the height limiting. now its being controlled just by the width limit. I can revert it to height limiting too, but it will make it get smaller as you saw

lelemm commented:
VRT done after master merge @MikesGlitch

youngcw commented:
The text still needs some help.  
![image](https://github.com/user-attachments/assets/45907507-efdd-4777-b867-2c5640e5cef9)

lelemm commented:
I will bring back the height limiting ok?

Jonathan-Fang commented:
This is a great report to add!

joel-jeremy commented on packages/desktop-client/src/components/reports/reports/SummaryCard.tsx:
```diff
@@ -0,0 +1,154 @@
+import { SummaryNumber } from '../SummaryNumber';
+import { useReport } from '../useReport';
+
+type SummaryProps = {
```
```suggestion
type SummaryCardProps = {
```

lelemm replied:
done

joel-jeremy reviewed (commented):
Just a few comments. But overall LGTM!

lelemm commented on packages/desktop-client/src/components/reports/SummaryNumber.tsx:
```diff
@@ -0,0 +1,145 @@
+  const refDiv = useRef<HTMLDivElement>(null);
+  const offScreenRef = useRef<HTMLDivElement>(null);
+
+  const FONT_SIZE_SCALE_FACTOR = 0.9;
```
done

lelemm commented on packages/desktop-client/src/components/reports/SummaryNumber.tsx:
```diff
@@ -0,0 +1,145 @@
+import { chartTheme } from './chart-theme';
+import { LoadingIndicator } from './LoadingIndicator';
+
+type AnimatedNumberProps = {
```
done

lelemm commented on packages/desktop-client/src/icons/custom/CloseParenthesis.svg:
done

MikesGlitch reviewed (approved):
LGTM - looking for 1 more review before we merge
